DEPRECATIONS.md

```diff
@@ -0,0 +1,66 @@
+# Deprecations & Breaking Changes in KEDA
+
+This document describes how KEDA handles deprecations and breaking changes and should provide clarity to KEDA end-users on what to expect when upgrading to a new version of KEDA.
+
+If you are interested in the current open deprecations, we recommend checking our [GitHub Discussions](https://github.com/kedacore/keda/discussions/categories/deprecations) or the open [breaking changes](https://github.com/kedacore/keda/issues?q=is%3Aopen+label%3Abreaking-change+sort%3Aupdated-desc).
+
+## Our versioning strategy
+
+KEDA follows the deprecation strategy of Kubernetes and is allowed to make breaking changes in minor versions after a deprecation period.
+
+The KEDA runtime, which consists of the operator and metric server, will provide support on the deprecated features for 4 releases/1 year after the deprecation was announced. In the following release, the feature will be removed.
+
+Depending on the area of the change, KEDA might not wait 4 releases to introduce a breaking change because one of its dependencies such as Kubernetes or scaler dependencies. However, these are the only exceptions to the rule.
+
+### Custom Resource Definitions (CRDs)
+
+Our custom resource definitions (CRDs) have a separate versioning scheme than the KEDA runtime given they have an `apiVersion`. For this, we follow the [official Kubernetes API versioning policy](https://kubernetes.io/docs/reference/using-api/#api-versioning).
+
+KEDA is allowed to make breaking changes to the CRDs, when a new `apiVersion` is introduced.
```

Comment:
```suggestion
KEDA is allowed to make breaking changes to the CRDs, when a new `apiVersion` is introduced. Any of those breaks would be coupled with a KEDA versioned release, with deprecations of previous `apiVersion` held for major version releases.
```

LMK if this is accurate (esp. the waiting to deprecate old api versions with major versions

Reply:
This is actually not correct, I think as the api versions will be removed in the next minor version.

Thoughts @zroubalik @JorTurFer ?

Reply:
Isn't deprecating a previously supported configuration a breaking change?

Reply:
Oh wait I'm sorry this is about KEDA keeping up with K8s APIs? There it makes sense we'd keep up with their latest versions which are gated behind semver